### ISSUE TITLE
Adding_Scheduler

### ISSUE THREAD

**Issue by tridelt** [tridelt/PyTorchWord2Vec#3]
https://github.com/tridelt/PyTorchWord2Vec/blob/8e89d89c1305e9582395df2ffb180afdab7f5bf6/train.ipynb#L76

Hab den [Cosine-Scheduler](https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.CosineAnnealingLR) so implementiert wie du es mir gesagt hast und es funktioniert! Danke dir.

Ich habe die anfängliche LR von 0.001 auf 0.1 geändert und das Modell lernt nun in 5 Epochen wofür ich vorher 600 Epochen gebraucht habe. Leider senkt der Scheduler die LR nicht schnell genug, sodass wenn der validate-Error stagniert das Minimum mit einer kleineren LR vorsichtiger erkundet werden kann.
<img width="1060" alt="screenshot 2019-03-01 at 10 56 31" src="https://user-images.githubusercontent.com/24197322/53630590-ae203380-3c10-11e9-81ab-42dd56baab6f.png">

Wäre es nicht sinnvoll einen Scheduler zu wählen welchem man für jedem Step auch den Validate-Loss senden müsste damit der Scheduler weiss ob er jetzt schon eine kleinere LR wählen soll oder habe ich den Cosine-Scheduler falsch eingestellt? Dieser verringert bei mir nämlich viel zu langsam die LR.
